shanoir-ng-studies/src/main/java/org/shanoir/ng/study/service/StudyServiceImpl.java

```diff
@@ -184,11 +189,21 @@ public Study update(final Study study) throws EntityNotFoundException, MicroServ
 				subjectStudy.setStudy(studyDb);
 			}
 		}
-		
-		if (study.getProtocolFilePaths() != null) {
-			studyDb.setProtocolFilePaths(study.getProtocolFilePaths());
+
+		if (studyDb.getProtocolFilePaths() != null) {
+			for(String filePath : studyDb.getProtocolFilePaths()) {
+				if (!study.getProtocolFilePaths().contains(filePath)) {
+					// Delete file
+					String filePathToDelete = getStudyFilePath(studyDb.getId(), filePath);
+					System.err.println("COCUOCUOCUCOCUCUCOUCOCCUO    " + filePathToDelete);
```

Comment:
Hi Jean-Come, not sure if this big cocuocuocuo is ready for prod :-))))

Reply:
Oupsi ! I'm trying to go too fast again !